### DISCUSSION TITLE
升级到v2.0.10版本中百度网盘webdav代理问题

### DISCUSSION THREAD

**Discussion by zyipk** [AlistGo/alist#606] [Q&A] [answered]
在升级到v2.0.10版本后，我在使用webdav服务，如果没有开启webdav_direct中转的话，无法用播放器在线观看百度网盘的视频。

在v2.0.9中，不用代理中转，也可以使用webdav在线观看百度网盘的视频，我使用webdav下载时，下载速度取决于宽带的下行速度。

在v2.0.10版本使用，就需要我开启中转，否则不能使用，这样的话速度将受限于我的服务器的上传宽带。

客户端是使用安卓的Nplayer。

**Comment by xhofe** [comment, marked as answer]
2.0.9->2.0.10并无改动。且测试2.0.9同样无法播放。

**Reply by zyipk** [reply]
> 2.0.9->2.0.10并无改动。且测试2.0.9同样无法播放。

我换了个新的服务器搭建了Alist 并测试了v2.0.9和v2.0.10版本，v2.0.9确实能正常播放，而v2.0.10不能。

![image](https://user-images.githubusercontent.com/44421580/155642394-1806e842-2ad6-4763-baf0-c33bfbe7017c.png)

**Reply by xhofe** [reply]
https://github.com/Xhofe/alist/compare/v2.0.9...v2.0.10
对百度网盘唯一的改动就是：

2.0.9 在请求重定向地址时未设置user agent导致重定向失败，在返回的url中会包含access_token，会导致安全问题。在2.0.10中修复了这个问题。其他无任何变动。
且根据百度网盘开放平台的文档，不携带`user-agent: pan.baidu.com`是无法下载的：https://pan.baidu.com/union/doc/pkuo3snyp

**Reply by zyipk** [reply]
> [v2.0.9...v2.0.10](https://github.com/Xhofe/alist/compare/v2.0.9...v2.0.10) 对百度网盘唯一的改动就是：
> 
> 2.0.9 在请求重定向地址时未设置user agent导致重定向失败，在返回的url中会包含access_token，会导致安全问题。在2.0.10中修复了这个问题。其他无任何变动。 且根据百度网盘开放平台的文档，不携带`user-agent: pan.baidu.com`是无法下载的：https://pan.baidu.com/union/doc/pkuo3snyp

确实是改了这个导致我使用webdav时无法播放视频，我fork了项目，改回去后，重新编译后能播放了，但这个有安全性问题，不过自用的话，问题应该不大吧。

**Reply by zyipk** [reply]
> 2.0.9->2.0.10并无改动。且测试2.0.9同样无法播放。

重新编译后，我也测试了一下，在Windows下，我使用RaiDrive挂载webdav，可以使用播放器播放视频，在安卓下，使用nPlayer和Reex也均可连接webdav观看百度网盘的视频，好奇怪。

**Reply by 839781960** [reply]
那就不用3.2了用209
